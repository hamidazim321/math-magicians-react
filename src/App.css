.Home {
  display: flex;
  padding: 2%;
  align-items: center;
}
