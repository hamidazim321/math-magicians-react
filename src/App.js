import './App.css';

function App() {
  return (
    <div>
      new code
    </div>
  );
}

export default App;
