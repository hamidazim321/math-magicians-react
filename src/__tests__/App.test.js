import {
  render, screen, fireEvent, cleanup,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../App';

afterEach(() => {
  cleanup();
});

describe('Integration test for App', () => {
  test('Renders App', () => {
    	const tree = renderer
    		.create(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
      )
    		.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('load the Quotes component by clickin on quotes in Navbar', async () => {
    const response = [
      {
        quote: 'mocked quote',
      },
    ];
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue(response),
    });

    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    const quotes = await screen.findByText(/Quotes/);
    fireEvent.click(quotes);

    expect((await screen.findByText(/mocked/)).textContent).toBe('mocked quote');
  });

  test('load the calculator component by clickin on Calculator in Navbar', async () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    const calculator = await screen.findByText(/Calculator/);
    fireEvent.click(calculator);

    const calc = document.getElementsByClassName('calculator')[0];
    expect(calc).toBeInTheDocument();
  });
});
