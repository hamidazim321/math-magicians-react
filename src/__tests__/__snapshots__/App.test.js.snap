// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test for App Renders App 1`] = `
Array [
  <nav>
    <p
      className="brandName"
    >
      Math Magicians
    </p>
    <ul>
      <li>
        <a
          aria-current="page"
          className="links active"
          href="/"
          onClick={[Function]}
        >
           Home 
        </a>
      </li>
      <li>
        <a
          className="links"
          href="/calculator"
          onClick={[Function]}
        >
           Calculator 
        </a>
      </li>
      <li>
        <a
          className="links"
          href="/quotes"
          onClick={[Function]}
        >
           Quotes 
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="Front"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sed ante vel enim molestie volutpat. Sed non lacus vel odio pretium sodales vitae id purus. Nulla facilisi. Ut porta fringilla ante, quis porta est sagittis eu. Nunc bibendum libero quis neque malesuada malesuada. Sed pellentesque auctor lacus a dictum. Proin blandit fermentum consectetur. Pellentesque molestie, nulla quis lacinia placerat, lectus elit pharetra mi, eu hendrerit ante ante eu felis. Nulla laoreet, tortor in placerat bibendum, dui purus semper augue, ut faucibus libero magna id tellus.
    </p>
    <p>
      Vestibulum at velit in neque finibus pellentesque. Donec sodales eget turpis in condimentum. Aenean malesuada erat odio, non venenatis ligula vulputate eget. Sed viverra nisl in libero bibendum vehicula. In bibendum metus et magna placerat ornare. Sed porttitor bibendum risus, nec tristique tortor tincidunt in. Nam nibh erat, finibus et pellentesque non, ultricies eget nisi. Nunc luctus vulputate urna, at tincidunt libero.
    </p>
  </div>,
]
`;
